hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/cache/CacheManager.java

```diff
@@ -82,64 +85,91 @@ public void init() {
                 publishCache.clear();
             }
             msgLog.infoInitPublishCache();
-            updateActiveIds();
+            requestCacheUpdate();
+
             definitions.registerListener(e -> {
-                updateActiveIds();
-            }, CONDITION_CHANGE, TRIGGER_REMOVE);
+                requestCacheUpdate();
+            }, TRIGGER_CONDITION_CHANGE, TRIGGER_REMOVE);
+
         } else {
             msgLog.warnDisabledPublishCache();
         }
     }
 
-    private synchronized void updateActiveIds() {
+    // Just run updateCache one time if multiple requests come in while an update is already in progress...
+    private void requestCacheUpdate() {
+        log.debug("Cache update requested");
+        if (updateRequested) {
+            log.debug("Cache update, redundant request ignored.");
+            return;
+        }
+
+        updateRequested = true;
```

Comment:
I have one minor doubt, non blocker about if as we pass this object as a listener, if the object is invoked in the tx context of the EJB.
Basically, the listeners are invoked on CassDefinitionsServiceImpl, so they are using a ejb pool for that.
But as the object is passed in "raw" (without container visibility) not sure if this method is invoked with the Lock(WRITE) semantic or not.
In case it is not, several threads might invoke it and modify the updateRequested variable, so on that case perhaps it should be marked as volatile.
But, I'm not 100% sure.
Just leaving this note here.